### PR TITLE
Refactor type `Environment` and env `Config` to support  new type`klient.Client`

### PR DESCRIPTION
**Pull request by vladimirvivien**
This PR is needed to refactor the way type `Environment` is constructed and initialized with package `klient` which did not exist when this package was created.  This work is also a precursor to the refactoring work that will take place to fix context propagation.

This potentially will address

*  #31 
*  #32

Context propagation will be handled in a future PR

### CODE REVIEW

**Comment by vladimirvivien** [comment]
@cpanato @ShwethaKumbla  @andrewsykim  PTAL

File: klient/client.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	return New(cfg)
+}
+
+func (c *client) RESTConfig() *rest.Config {
+	return c.cfg
+}
+
+func (c *client) Resources() *resources.Resources {
+	return c.resources
+}
```
Comment (cpanato) [inline klient/client.go:61]: nit: new line

File: pkg/envconf/config.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+// Labels returns the environment's label filters
+func (c *Config) Labels() map[string]string {
+	return c.labels
+}
+
+func randNS() string {
+	bytes := make([]byte, 25)
+	rand.Read(bytes)
+	return fmt.Sprintf("testns-%s", hex.EncodeToString(bytes))
+}
```
Comment (cpanato) [inline pkg/envconf/config.go:153]: nit: new line

**Review by cpanato — COMMENTED** [review]
thanks, lgtm, just need to fix the lint issues we I think we are good to go

File: pkg/envconf/config.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	return c
+}
+
+// Labels returns the environment's label filters
+func (c *Config) Labels() map[string]string {
+	return c.labels
+}
+
+func randNS() string {
+	bytes := make([]byte, 25)
```
Comment (cpanato) [inline pkg/envconf/config.go:150]: maybe we can use `github.com/google/uuid` and generate an uuid and remove the `testns-%s`.

just an idea :)

Comment (cpanato) [reply pkg/envconf/config.go:150]: make sense, as i said just a suggestion :)

**Pull request merged**